docs/guide/custom-functions.md

```diff
@@ -119,72 +125,116 @@ Register your function plugin (and its translations) so that HyperFormula can re
 };
 
-HyperFormula.registerFunctionPlugin(CountHF, CountHF.translations);
+HyperFormula.registerFunctionPlugin(GreetingsPlugin, GreetingsPlugin.translations);
+```
+
+### Test your function
+
+You may add some unit tests to make sure that your custom function works correctly using one of the javascript testing libraries.
+
+```js
+describe('GreetingsPlugin', () => {
+  it('works for a non-empty string', () => {
+    HyperFormula.registerFunctionPlugin(GreetingsPlugin, GreetingsPlugin.translations);
+    const engine = HyperFormula.buildFromArray([['Anthony', '=GREET(A1)']], { licenseKey: 'gpl-v3' });
+    expect(engine.getCellValue({ sheet: 0, row: 0, col: 1 })).toEqual('👋 Hello, Anthony!');
+  });
+
+  it('returns #VALUE! error for empty string', () => {
+    HyperFormula.registerFunctionPlugin(GreetingsPlugin, GreetingsPlugin.translations);
+    const engine = HyperFormula.buildFromArray([['', '=GREET(A1)']], { licenseKey: 'gpl-v3' });
+    expect(engine.getCellValueType({ sheet: 0, row: 0, col: 1 })).toEqual('ERROR');
+    expect(engine.getCellValue({ sheet: 0, row: 0, col: 1 }).value).toEqual('#VALUE!');
+  });
+});
 ```
 
 ## Custom function options
 
-You can configure your custom function to:
-* Use the [array arithmetic mode](arrays.md).
-* Treat reference or range arguments as arguments that don't create dependency.
-* Inline range arguments to scalar arguments.
-* Get recalculated with each sheet shape change.
-* Be a [volatile](volatile-functions.md) function.
-* Repeat a specified number of last arguments indefinitely.
-* Never get vectorized.
+You can set the following options for your function:
+
+| Option                              | Type    | Description                                                                                                                                  |
+|-------------------------------------|---------|----------------------------------------------------------------------------------------------------------------------------------------------|
+| `method` (required)                 | String  | Name of the method that implements the custom function logic.                                                                                |
+| `parameters`                        | Array   | Specification of the arguments accepted by the function and their [validation options](#argument-validation-options).                        |
+| `repeatLastArgs`                    | Number  | For functions with a variable number of arguments: sets how many last arguments can be repeated indefinitely.                                |
+| `expandRanges`                      | Boolean | `true`: ranges in the function's arguments are inlined to (possibly multiple) scalar arguments.                                              |
+| `returnNumberType`                  | String  | Function return type.                                                                                                                        |
```

Comment:
A list of possible values would be nice.

Reply:
Added